src/WorkerFarm.js

```diff
@@ -86,4 +87,13 @@ for (let key in EventEmitter.prototype) {
   WorkerFarm.prototype[key] = EventEmitter.prototype[key];
 }
 
+function getNumWorkers() {
+  const cores = os.cpus().filter(function(cpu, index) {
+    const hasHyperthreading = cpu.model.includes('Intel');
```

Comment:
Is this really a good assumption? Do all Intel CPUs have hyperthreading?

Reply:
That's the else catchall branch of physical-cpu-count.

I don't know

Reply:
also i’m pretty sure AMD’s newer Ryzen cpus have SMT, which is what Intel calls hyperthreading

Reply:
[Here](https://gist.github.com/brandon93s/a46fb07b0dd589dc34e987c33d775679) is a Gist of how it's performed by [physical-cpu-count](https://www.npmjs.com/package/physical-cpu-count).